web/src/components/organisms/Project/Schema/hooks.ts

```diff
@@ -16,10 +19,30 @@ type Params = {
 
 export default ({ projectId, modelId }: Params) => {
   const [modelModalShown, setModelModalShown] = useState(false);
-  const [fieldModalShown, setFieldModalShown] = useState(false);
+  const [fieldCreationModalShown, setFieldCreationModalShown] = useState(false);
+  const [fieldUpdateModalShown, setFieldUpdateModalShown] = useState(false);
+  const [isKeyAvailable, setIsKeyAvailable] = useState(false);
+  const [selectedField, setSelectedField] = useState<Field | null>();
+  const [selectedType, setSelectedType] = useState<FieldType | null>(null);
```

Comment:
```suggestion
  const [selectedField, setSelectedField] = useState<Field>(null);
  const [selectedType, setSelectedType] = useState<FieldType>(null);
```

Reply:
@KaWaite , I think this won't work
It gives me `Argument of type 'null' is not assignable to parameter of type 'Field `

Reply:
Oh sorry, okay then
```suggestion
  const [selectedField, setSelectedField] = useState<Field | null>(null);
  const [selectedType, setSelectedType] = useState<FieldType | null>(null);
```